src/components/auth/SignUp.vue

```diff
@@ -26,117 +41,119 @@
           :class="buttonStatusClass"
           id="generate-button"
           @click="submitInformation"
-        >Create Account and Generate an API Key</button>
+        >
+          Create Account and Generate an API Key
+        </button>
       </div>
     </div>
     <a
       class="has-text-dark has-text-underlined"
       @click="$emit('toggle-has-account', true)"
-    >Already have an account?</a>
+      >Already have an account?</a
+    >
   </div>
 </template>
 
 <script>
-  import { HTTP } from "../../http-common";
+import { HTTP } from "../../http-common";
 
-  export default {
-    data() {
-      return {
-        buttonStatusClass: "",
-        username: "",
-        password: "",
-        passwordConfirm: "",
-      };
+export default {
+  data() {
+    return {
+      buttonStatusClass: "",
+      username: "",
+      password: "",
+      passwordConfirm: ""
+    };
+  },
+  methods: {
+    async submitInformation() {
+      const usernameRegex = /^[a-zA-Z0-9-]+$/;
+      if (this.username === "" || this.username.length < 4) {
+        this.$emit(
+          "new-error",
+          "You must choose a username that is greater than 4 characters."
+        );
+        this.buttonStatusClass = "";
+        return;
+      }
+      if (this.username.match(usernameRegex) == null) {
+        this.$emit(
+          "new-error",
+          "Your username can only include alphanumeric characters and '-'"
+        );
+        this.buttonStatusClass = "";
+        return;
+      }
+      if (this.password === "" || this.password.length < 8) {
+        this.$emit(
+          "new-error",
+          "You must choose a password that is greater than 8 characters."
+        );
+        this.buttonStatusClass = "";
+        return;
+      }
+      if (this.passwordConfirm === "") {
+        this.$emit("new-error", "You must confirm your password.");
+        this.buttonStatusClass = "";
+        return;
+      }
+      if (this.password !== this.passwordConfirm) {
+        this.$emit("new-error", "Your passwords do not match.");
+        this.buttonStatusClass = "";
+        return;
+      }
+      if (this.username === this.password) {
+        this.$emit(
+          "new-error",
+          "You can't use the same username and password."
+        );
```

Comment:
Spacing between blocks would be great.

Reply:
Added now. But I think for the future it would be good separating those into multiple functions, or even mixins.